pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisMessageBatch.java

```diff
@@ -58,4 +68,9 @@ public StreamPartitionMsgOffset getOffsetOfNextBatch() {
   public boolean isEndOfPartitionGroup() {
     return _endOfShard;
   }
+
+  @Override
+  public boolean hasMissingOffsets() {
+    return MessageBatch.super.hasMissingOffsets();
```

Comment:
Let's also add a TODO to support detecting data loss for Kinesis
```suggestion
    return false;
```

Reply:
Actually we can simply not override it